From a5978074dba200b4059f53ab94b9c34863250520 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Tue, 3 Mar 2015 12:25:39 +0000
Subject: Convert from RRSetChanged() to RROutputChanged()

RRSetChanged() is much more recent than RROutputChanged(), and
RROutptChanged() allows for more finer grained updates. Switch to
RROutputChanged for compilation with xorg-server-1.14 and older.

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>

diff --git a/src/sna/sna_display.c b/src/sna/sna_display.c
index c3cc6ba..cc82806 100644
--- a/src/sna/sna_display.c
+++ b/src/sna/sna_display.c
@@ -4103,6 +4103,7 @@ reset:
 				goto cleanup;
 		}
 
+		RROutputChanged(output->randr_output, TRUE);
 		sna_output_create_resources(output);
 		RRPostPendingProperties(output->randr_output);
 
@@ -4314,6 +4315,7 @@ void sna_mode_discover(struct sna *sna)
 		} else {
 			sna_output->id = 0;
 			output->crtc = NULL;
+			RROutputChanged(output->randr_output, TRUE);
 		}
 		changed |= 2;
 	}
@@ -6167,7 +6169,7 @@ sna_mode_set_primary(struct sna *sna)
 
 		DBG(("%s: setting PrimaryOutput %s\n", __FUNCTION__, output->name));
 		rr->primaryOutput = output->randr_output;
-		RROutputChanged(rr->primaryOutput, 0);
+		RROutputChanged(rr->primaryOutput, FALSE);
 		rr->layoutChanged = TRUE;
 		break;
 	}
diff --git a/src/uxa/intel_display.c b/src/uxa/intel_display.c
index 45afbae..544382a 100644
--- a/src/uxa/intel_display.c
+++ b/src/uxa/intel_display.c
@@ -1493,6 +1493,7 @@ intel_output_init(ScrnInfoPtr scrn, struct intel_mode *mode, drmModeResPtr mode_
 			intel_output = output->driver_private;
 			intel_output->output_id = mode_res->connectors[num];
 			intel_output->mode_output = koutput;
+			RROutputChanged(output->randr_output, TRUE);
 			return;
 		}
 	}
@@ -2495,6 +2496,7 @@ restart_destroy:
 		drmModeFreeConnector(intel_output->mode_output);
 		intel_output->mode_output = NULL;
 		intel_output->output_id = -1;
+		RROutputChanged(output->randr_output, TRUE);
 
 		changed = TRUE;
 		if (mode->delete_dp_12_displays) {
@@ -2525,10 +2527,8 @@ restart_destroy:
 		intel_output_init(scrn, intel->modes, mode_res, i, 1);
 	}
 
-	if (changed) {
-		RRSetChanged(xf86ScrnToScreen(scrn));
+	if (changed)
 		RRTellChanged(xf86ScrnToScreen(scrn));
-	}
 
 	drmModeFreeResources(mode_res);
 out:
-- 
cgit v0.10.2

